packages/dx-grid-core/src/plugins/grid-exporter/computeds.ts

```diff
@@ -0,0 +1,151 @@
+import {
+  OutlineLevelsFn, FilterSelectedRowsFn, GetRowsToExportFn, Row, BuildGroupTreeFn,
+  GetExportSummaryFn, GetCloseGroupFn, Grouping,
+} from '../../types';
+import { PureComputed } from '@devexpress/dx-core';
+import { ROOT_GROUP } from './constants';
+import { exportSummaryItems, removeEmptyGroups } from './helpers';
+import { TABLE_DATA_TYPE } from '../table/constants';
+
+export const groupOutlineLevels: OutlineLevelsFn = grouping => (
+  grouping?.reduce((acc, { columnName }, index) => ({
+    ...acc,
+    [columnName]: index,
+  }), {}) || {}
+);
+
+const filterSelectedRows: FilterSelectedRowsFn = (rows, selection, getRowId, isGroupRow) => {
+  const selectionSet = new Set<any>(selection);
+  return rows.filter(row => (
+    isGroupRow && isGroupRow(row)) || selectionSet.has(getRowId(row)),
+  );
+};
+
+export const rowsToExport: GetRowsToExportFn = (
+  rows, selection, grouping, getCollapsedRows, getRowId, isGroupRow,
+) => {
+  const expandRows: PureComputed<[Row[]]> = collapsedRows => (
+    collapsedRows.reduce((acc, row) => (
+      [...acc, row, ...(expandRows(getCollapsedRows(row) || []))]
+    ), [])
+  );
+
+  const expandedRows = getCollapsedRows ? expandRows(rows) : rows;
+
+  if (!!selection) {
+    const filteredRows = filterSelectedRows(expandedRows, selection, getRowId, isGroupRow);
+    return removeEmptyGroups(filteredRows, grouping, isGroupRow);
+  }
+  return expandedRows;
```

Comment:
selection && return expandedRows;

Reply:
Seems I don't understand what you mean :)

Reply:
```js
if (selection) return expandedRows;
  
const filteredRows = filterSelectedRows(expandedRows, selection, getRowId, isGroupRow);
return removeEmptyGroups(filteredRows, grouping, isGroupRow);
```